editor/editor_interface.cpp

```diff
@@ -263,6 +265,77 @@ void EditorInterface::set_current_feature_profile(const String &p_profile_name)
 	EditorFeatureProfileManager::get_singleton()->set_current_profile(p_profile_name, true);
 }
 
+// Editor dialogs.
+
+void EditorInterface::popup_dialog_scene_tree(const Callable &p_callback, const TypedArray<StringName> &p_valid_types) {
+	if (dialog_scene_tree) {
+		dialog_scene_tree->queue_free();
+	}
+	dialog_scene_tree = memnew(SceneTreeDialog);
+	get_base_control()->add_child(dialog_scene_tree);
+
+	Vector<StringName> valid_types;
+	int length = p_valid_types.size();
+	valid_types.resize(length);
+	for (int i = 0; i < length; i++) {
+		valid_types.push_back(p_valid_types[i]);
+	}
+	dialog_scene_tree->set_valid_types(valid_types);
+
+	dialog_scene_tree->popup_scenetree_dialog();
+
+	Callable selected_callback = callable_mp(this, &EditorInterface::_node_selected).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("selected"), selected_callback);
+
+	Callable canceled_callback = callable_mp(this, &EditorInterface::_node_selection_canceled).bind(p_callback);
+	dialog_scene_tree->connect(SNAME("canceled"), canceled_callback);
+}
+
+void EditorInterface::_node_selected(const NodePath &p_node_path, const Callable &p_callback) {
+	TypedArray<NodePath> node_paths;
+	NodePath path = get_edited_scene_root()->get_path().rel_path_to(p_node_path);
```

Comment:
```suggestion
	NodePath path = get_edited_scene_root()->get_path_to(p_node_path);
```

Reply:
`get_path_to` takes a Node, not a NodePath. Is there a different way I should be getting the path here?

Reply:
Right, I missed that.